sdk/python/kfp/dsl/_metadata.py

```diff
@@ -153,6 +154,7 @@ def _extract_component_metadata(func):
     arg_type = None
     arg_default = arg_defaults[arg] if arg in arg_defaults else None
     if isinstance(arg_default, PipelineParam):
+      warnings.warn('Explicit creation of `kfp.dsl.PipelineParam`s by the users is deprecated. The users should define the parameter type and default values using standard pythonic constructs: def my_func(a: int = 1, b: str = "default"):')
```

Comment:
Shall we provide a way to suppress this warning, like a flag --verbose, or it needs to go out anyway?

Reply:
I'm not sure we need a way to disable it since the pipeline definitions did not need PipelineParams since the first public release and it's very easy to fix the issue (and make the pipeline function signature much smaller and nicer in the process).